### PR TITLE
Use accept header to choose metrics export format

### LINKED ISSUE #30343: Trailing comma is lost from prometheus metrics
### Describe the bug

In all released versions (including 2.15.3.Final), prometheus metrics (accessible at 'q/metrics') has format of json object with a trailing comma. In the current main branch (tested on dc85cd0d16c44a264507529c9ee26e26c7ffe175) the format is changed, and the comma was lost. Is this an expected change? Prometheus format allows[1] for both options, but that change may broke parsing of logs and therefore it would be nice to have it documented

[1] https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators

### Expected behavior

Example:
`http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/hello",} 1.0`

### Actual behavior

`http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/hello"} 6.0`

### How to Reproduce?

1. `mvn io.quarkus.platform:quarkus-maven-plugin:2.15.3.Final:create     -DprojectGroupId=org.acme     -DprojectArtifactId=micrometer-quickstart     -Dextensions='resteasy-reactive,micrometer-registry-prometheus'`
2. `cd  micrometer-quickstart`
3. `mvn quarkus:dev -Dquarkus.platform.version=999-SNAPSHOT`
4. `curl http://localhost:8080/hello/`
5. `curl http://localhost:8080/q/metrics | grep http_server_requests_seconds_count # example, many (all?) other metrics  also changed`

for comparision, replace step 3 with `mvn quarkus:dev -Dquarkus.platform.version=2.15.3.Final`

### Output of `uname -a` or `ver`

6.0.18-300.fc37.x86_64

### Output of `java -version`

17.0.4, vendor: GraalVM Community

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

dc85cd0d16c44a264507529c9ee26e26c7ffe175

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jtama**
Fixes #30343

### CODE REVIEW

**Comment by jtama** [comment]
To be added to the notes somewhere : 

Starting from now on (replace with exact 2.15 and 2.16 versions :grinning:) the format for /q/metrics endpoint response has switched to application/openmetrics-text, which provides support for exemplars. This format does have some differences, most notably that trailing commas have been removed. You can check the full specification here.
Fallback to the older format (text/plain) can be obtain by setting the `Accept` header?

**Review by ebullient — APPROVED** [review]

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/handlers/PrometheusHandler.java [hunk 1]
```diff
@@ -34,22 +34,25 @@ public void handle(RoutingContext routingContext) {
                     .setStatusMessage("Unable to resolve Prometheus registry instance");
         } else {
             ManagedContext requestContext = Arc.container().requestContext();
+            var acceptHeader = routingContext.request().getHeader("Accept").equals("text/plain") ? TextFormat.CONTENT_TYPE_004
```
Comment (gastaldi) [inline extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/handlers/PrometheusHandler.java:37]: `routingContext.request().getHeader("Accept")` can return `null`

Comment (jtama) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/handlers/PrometheusHandler.java]: Corrected the point. Could you please alow the build to pass again ?

**Comment by geoand** [comment]
@ebullient I see you have added the `noteworthy-feature` flag, but I am wondering whether it's applicable in this case as this appears to be a bug fix, or did I miss something?

**Comment by jtama** [comment]
> @ebullient I see you have added the `noteworthy-feature` flag, but I am wondering whether it's applicable in this case as this appears to be a bug fix, or did I miss something?

I think the main goal is to warn users that the default format as changed, and how they can revert to the plain/text format

**Comment by geoand** [comment]
Then is should be `breaking-change`, right?

**Comment by jtama** [comment]
Well i guess it should have been yes...

**Comment by geoand** [comment]
Up to @ebullient 🙂

**Comment by ebullient** [comment]
I forgot about breaking change. 🤦‍♀️

**Comment by geoand** [comment]
No worries 🙂

**Pull request merged**